models/issues/label.go

```diff
@@ -149,15 +149,26 @@ func (l *Label) LoadSelectedLabelsAfterClick(currentSelectedLabels []int64, curr
 		} else if s != 0 {
 			// Exclude other labels in the same scope from selection
 			if s < 0 || labelScope == "" || labelScope != currentSelectedExclusiveScopes[i] {
-				labelQuerySlice = append(labelQuerySlice, strconv.FormatInt(s, 10))
+				labelQuerySlice = append(labelQuerySlice, s)
 			}
 		}
 	}
+
 	if !labelSelected {
-		labelQuerySlice = append(labelQuerySlice, labelID)
+		labelQuerySlice = append(labelQuerySlice, l.ID)
 	}
 	l.IsSelected = labelSelected
-	l.QueryString = strings.Join(labelQuerySlice, ",")
+
+	// Sort and deduplicate the ids to avoid the crawlers asking for the
+	// same thing with simply a different order of parameters
+	slices.Sort(labelQuerySlice)
+	labelQuerySlice = slices.Compact(labelQuerySlice)
```

Comment:
Just use `code.gitea.io/gitea/modules/container.Set`. Then you don't need to sort and compact.

Reply:
Thanks for the review ! However, I used container.Set in [my previous iteration ](https://github.com/xlii-chl/gitea/blob/5314cd1202a2c47a53bb0174b2417f658f721e42/models/issues/label.go#L151)and the code was less readable and the output's ordering was random (container.Set uses map).
I still use it in [models/issues/issue_search.go](https://github.com/go-gitea/gitea/pull/26460/files#diff-59d96577a83fb843b6b65a0777998803eba3212a88e37211d1b13b24f285134f) where its `Add()` method and the on-the-fly de-duplication are quite handy.

Reply:
Didn't see that the output is not just used in the backend. Then you could just use the set with `slices.Sort` at the end but that is only better if `labelQuerySlice` is very large.

Reply:
I think KN4CK3R 's suggestion is right. Managed to do some rewriting: #31497